src/prefect/_internal/concurrency/services.py

```diff
@@ -46,7 +47,18 @@ def start(self):
             self.send(self._early_items.popleft())
 
         # Stop at interpreter exit by default
-        atexit.register(self.drain_all)
+        if sys.version_info < (3, 9):
+            atexit.register(self.drain)
+        else:
+            # See related issue at https://bugs.python.org/issue42647
```

Comment:
🤯

Reply:
This copied over from the old log worker. I had a great time with that one originally :D